src/webfaf/static/js/custom.js

```diff
@@ -64,6 +64,10 @@ $(document).ready(function() {
     });
 
     $('input.daterange').daterangepicker({
+      locale: {
+        // Set Monday as the first day of the week.
+        firstDay: 1
```

Comment:
Mmm… Maybe not. We should start using Moment.js with locales and use https://momentjs.com/docs/#/i18n/locale-data/.

Reply:
Actually, we would *only* need to use Moment.js with locales, as daterangepicker.js consults it for us.

Reply:
I agree the UI should match the user's world view, but in this specific instance there's a slight twist – the weekly summaries as we store them assume Monday as the first day of the week. Letting the locale choose the first day might contradict this convention, so I'm not sure how beneficial it would be in in the end.

Reply:
So, with that in mind, the ability to choose the date range is just an exercise in futility?

https://github.com/abrt/faf/blob/5bc42f3c87ddae20ac01cb3d96e73f7a9c0d2770/src/webfaf/reports.py#L183

The above directly queries the database, so having the internal representation use ISO 8601 shouldn’t be an issue at all, given that `ReportHistoryWeekly` is already questionable in terms of usefulness (i.e. it only shows stats for calendar weeks, as opposed to letting me see trends in weekly intervals going back from *today*)<sup>*</sup>.

<sup>*</sup> - I wasn’t particularly fond of (or good at) statistics at school, so I cannot tell at all if what we show is useful or is just for the sake of having something.

Reply:
Well, the date range is exact when displaying daily values. It is aligned to Monday-starting weeks (as we store them in the DB) on weekly and months on monthly views.

I'm not sure, what you mean by "having the internal representation use ISO 8601 shouldn’t be an issue at all".

> it only shows stats for calendar weeks, as opposed to letting me see trends in weekly intervals going back from today

I understand your remark, but I _think_ it makes more sense that "weekly" mean calendar weeks rather than "7-day intervals going back from today".<sup>†</sup> Why not go forward by 7 days from the first day instead? The might be problems with months as well: Are they calendar months? Or 28/30/n-day intervals?

Calendar weeks starting with Mondays are technically unambiguous – of course, there are l10n differences among various regions.

And, naturally, in the end, the way we process, display and present data depends on the needs and preferences of whoever is _actually_ using ABRT Analytics for doing stuff and making decisions. Unfortunately, we currently don't have much feedback on this front.

† Putting aside the fact that there's a bug in the current UI where whatever dates the users selects in the date picker, they do not necessarily align with the actual weeks as they are stored/displayed when the page loads in a weekly view.

Reply:
Ah, right, I completely missed the dropdown for granularity (daily/weekly/monthly).

Reply:
Okay, whatever, let’s go with Mondays. I’m not willing to die on this hill.